glib2/test/test-variant.rb

```diff
@@ -24,4 +24,71 @@ class TestGLibVariant < Test::Unit::TestCase
                    [variant.type, variant.value])
     end
   end
+
+  sub_test_case ".parse" do
+    test "value: true" do
+      variant = GLib::Variant.parse("true")
+      assert_equal(true, variant.value)
+    end
+
+    test "value: float" do
+      variant = GLib::Variant.parse(Math::PI.to_s)
+      assert_equal(Math::PI, variant.value)
+    end
+
+    test "value: infinity" do
+      variant = GLib::Variant.parse("inf")
+      assert_equal(Float::INFINITY, variant.value)
+    end
+
+    test "value: int vector" do
+      variant = GLib::Variant.parse("[1, 2, 3]")
+      assert_equal([1, 2, 3], variant.value)
+    end
+
+    test "value: integer, type: UINT16" do
+      variant = GLib::Variant.parse("65535", GLib::VariantType::UINT16)
+      assert_equal(65535, variant.value)
+      assert_equal("q", variant.type.to_s)
+    end
+  end
+
+  sub_test_case "#variant_print" do
+    test "value: true" do
+      variant = GLib::Variant.new(true)
+      out = variant.variant_print
+      assert_equal("true", out)
+    end
+
+    test "value: float" do
+      variant = GLib::Variant.new(Math::PI)
+      out = variant.variant_print
+      ## this may avoid mismatch from rounding error
+      assert_equal(Math::PI, out.to_f)
+    end
+
+    test "value: infinity" do
+      variant = GLib::Variant.new(Float::INFINITY)
+      out = variant.variant_print
+      assert_equal("inf", out)
+    end
+
+    test "value: int vector" do
+      variant = GLib::Variant.new([1, 2, 3])
+      out = variant.variant_print
+      assert_equal("[1, 2, 3]", out)
+    end
+
+    test "value: integer, type: UINT16, type_annotate: true" do
```

Comment:
We can remove `type: UINT16, ` because it's not important in this test:

```suggestion
    test "integer, type_annotate: true" do
```